src/gui/qgsvariableeditorwidget.cpp

```diff
@@ -719,6 +719,8 @@ void VariableEditorDelegate::setModelData( QWidget *widget, QAbstractItemModel *
   {
     //edited variable name
     QString newName = lineEdit->text();
+    newName = newName.trimmed();
+    newName = newName.replace( QStringLiteral( " " ), "_" );
```

Comment:
Talking about wasted CPU, use the `replace( QChar, QChar )` overload ;)

Reply:
And I think this would be worth its own pull request.

Reply:
Thanks. Was meant to be in a different branch.